src/modal/modal.module.ts

```diff
@@ -1,8 +1,19 @@
 import {NgModule} from '@angular/core';
 
+import {NgbModalContainer} from './modal_container';
 import {NgbModalBackdrop} from './modal_backdrop';
 import {NgbModalWindow} from './modal_window';
+import {NgbModalStack} from './modal_stack';
+import {NgbModal} from './modal';
 
-@NgModule({declarations: [NgbModalBackdrop, NgbModalWindow], entryComponents: [NgbModalBackdrop, NgbModalWindow]})
+export {NgbModal, NgbModalOptions} from './modal';
+export {NgbModalRef} from './modal_ref';
+
+@NgModule({
+  declarations: [NgbModalContainer, NgbModalBackdrop, NgbModalWindow],
+  entryComponents: [NgbModalBackdrop, NgbModalWindow],
+  exports: [NgbModalContainer],
+  providers: [NgbModal, NgbModalStack]
```

Comment:
Yes, as I need to share the same instance between `NgbModal` and `NgbModalContainer`